app/ui/browser/views/navbar/navbar.jsx

```diff
@@ -229,4 +230,13 @@ NavBar.propTypes = {
   navigateTo: PropTypes.func.isRequired,
 };
 
-export default NavBar;
+const mapStateToProps = (state) => {
+  const browserWindow = state.browserWindow;
```

Comment:
So `<NavBar/>` only attempts to render if either `numPages` or `page` state changes? Nice nice nice